### PR TITLE
pd: set madvdontneed=1 to shrink RSS after GC

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 

speed up ram gc

### What is changed and how it works?

tidb has similar param: https://github.com/pingcap/tiup/blob/master/embed/templates/scripts/run_tidb.sh.tpl

### Check List 

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Comment by AstroProfundis** [comment]
@lhy1024 please sign the CLA so that we can merge this PR.

**Review by AstroProfundis — APPROVED** [review]

**Comment by AstroProfundis** [comment]
@lhy1024 It seems the CI tests are keep failing

**Comment by lhy1024** [comment]
> @lhy1024 It seems the CI tests are keep failing

There are some logs about this fail. How will I get them?

**Comment by lhy1024** [comment]
> > There are some logs about this fail. How will I get them?
> 
> You may try to find them in the assets of Github Actions run. The stdout shows that PD fails to start in certain tests.

Sorry, I am referring to how should I get the log that appears in the github action, such as `/home/tidb/deploy/pd-2379/log`, which appears `Error: failed to start pd: failed to start: n5 pd-2379.service, please check the instance's log(/home/tidb/ deploy/pd-2379/log) for more detail.: timed out waiting for port 2379 to be started after 2m0s` .

**Comment by AstroProfundis** [comment]
Ummm, I reviewed the workflow and seems the log uploading has been broken for long...

Maybe you can try to reproduce that on local server? Just follow the steps in `.github/workflows/integrate-cluster-cmd.yaml` should be fine.

File: embed/templates/scripts/run_pd.sh.tpl [hunk 1]
```diff
@@ -18,9 +18,9 @@ cd "${DEPLOY_DIR}" || exit 1
 {{- end}}
 
 {{- if .NumaNode}}
-exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} bin/pd-server \
+exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} GODEBUG=madvdontneed=1 bin/pd-server \
```
Comment (nexustar) [inline embed/templates/scripts/run_pd.sh.tpl:21]:
```suggestion
exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} env GODEBUG=madvdontneed=1 bin/pd-server \
```

File: embed/templates/scripts/run_pd.sh.tpl [hunk 2]
```diff
@@ -18,9 +18,9 @@ cd "${DEPLOY_DIR}" || exit 1
...
 {{- else}}
-exec bin/pd-server \
+exec GODEBUG=madvdontneed=1 bin/pd-server \
```
Comment (nexustar) [inline embed/templates/scripts/run_pd.sh.tpl:23]:
```suggestion
exec env GODEBUG=madvdontneed=1 bin/pd-server \
```

File: embed/templates/scripts/run_pd_scale.sh.tpl [hunk 3]
```diff
@@ -18,9 +18,9 @@ cd "${DEPLOY_DIR}" || exit 1
 {{- end}}
 
 {{- if .NumaNode}}
-exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} bin/pd-server \
+exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} GODEBUG=madvdontneed=1 bin/pd-server \
```
Comment (nexustar) [inline embed/templates/scripts/run_pd_scale.sh.tpl:21]:
```suggestion
exec numactl --cpunodebind={{.NumaNode}} --membind={{.NumaNode}} env GODEBUG=madvdontneed=1 bin/pd-server \
```

File: embed/templates/scripts/run_pd_scale.sh.tpl [hunk 4]
```diff
@@ -18,9 +18,9 @@ cd "${DEPLOY_DIR}" || exit 1
...
 {{- else}}
-exec bin/pd-server \
+exec GODEBUG=madvdontneed=1 bin/pd-server \
```
Comment (nexustar) [inline embed/templates/scripts/run_pd_scale.sh.tpl:23]:
```suggestion
exec env GODEBUG=madvdontneed=1 bin/pd-server \
```

**Pull request merged**